multichain-testing/test/fast-usdc/fast-usdc.test.ts

```diff
@@ -0,0 +1,295 @@
+  t.log('usdcDenom', usdcDenom);
+
+  await startContract(contractName, contractBuilder, {
+    oracle: keys(oracleMnemonics).map(n => `${n}:${wallets[n]}`),
+    usdcDenom: usdcDenom,
+    feedPolicy: makeFeedPolicy(nobleAgoricChannelId),
+    ...commonBuilderOpts,
+  });
+
+  // provide faucet funds for LPs
+  await faucetTools.fundFaucet([['noble', 'uusdc']]);
+
+  // save an LP in test context
+  const lpUser = await provisionSmartWallet(wallets['lp'], {
+    USDC: 100n,
+    BLD: 100n,
+  });
+
+  t.context = {
+    ...common,
+    lpUser,
+    oracleWds,
+    nobleAgoricChannelId,
+    usdcOnOsmosis,
+    wallets,
+  };
+});
+
+test.after(async t => {
+  const { deleteTestKeys } = t.context;
+  deleteTestKeys(accounts);
+});
+
+const toOracleOfferId = (idx: number) => `oracle${idx + 1}-accept`;
+
+test.serial('oracles accept', async t => {
+  const { oracleWds, retryUntilCondition, vstorageClient, wallets } = t.context;
+
+  const instances = await vstorageClient.queryData(
+    'published.agoricNames.instance',
+  );
+  const instance = fromEntries(instances)[contractName];
+
+  // accept oracle operator invitations
+  await Promise.all(
+    oracleWds.map(makeDoOffer).map((doOffer, i) =>
+      doOffer({
+        id: toOracleOfferId(i),
+        invitationSpec: {
+          source: 'purse',
+          instance,
+          description: 'oracle operator invitation', // TODO export/import INVITATION_MAKERS_DESC
+        },
+        proposal: {},
+      }),
+    ),
+  );
+
+  for (const name of keys(oracleMnemonics)) {
+    const addr = wallets[name];
+    await t.notThrowsAsync(() =>
+      retryUntilCondition(
+        () => vstorageClient.queryData(`published.wallet.${addr}.current`),
+        ({ offerToUsedInvitation }) => {
+          return offerToUsedInvitation[0][0] === `${name}-accept`;
+        },
+        `${name} invitation used`,
+      ),
+    );
+  }
+});
+
+test.serial('lp deposits', async t => {
+  const { lpUser, retryUntilCondition, vstorageClient } = t.context;
+
+  const lpDoOffer = makeDoOffer(lpUser);
+  const brands = await vstorageClient.queryData('published.agoricNames.brand');
+  const { USDC } = Object.fromEntries(brands);
+
+  const usdcGive = make(USDC, 10_000_000n);
+
+  const { shareWorth: currShareWorth } = await vstorageClient.queryData(
+    `published.${contractName}.poolMetrics`,
+  );
+
+  await lpDoOffer({
+    id: `lp-deposit-${Date.now()}`,
+    invitationSpec: {
+      source: 'agoricContract',
+      instancePath: [contractName],
+      callPipe: [['makeDepositInvitation']],
+    },
+    proposal: {
+      give: { USDC: usdcGive },
+      want: { PoolShare: divideBy(usdcGive, currShareWorth) },
+    },
+  });
+
+  await t.notThrowsAsync(() =>
+    retryUntilCondition(
+      () => vstorageClient.queryData(`published.${contractName}.poolMetrics`),
+      ({ shareWorth }) =>
+        !isGTE(currShareWorth.numerator, shareWorth.numerator),
+      'share worth numerator increases from deposit',
```

Comment:
I suppose this is reasonable for testing it worked. It would also be good to verify that the pool shares were received, but I think we can do that implicitly in the todo withdrawal test below.

Reply:
`doOffer` checks that `numWantsSatisfied` is 1, so we can have confidence the pool shares were received.

No harm in explicitly checking in the test though, so I fixed up to include that. Also closed out the `test.todo('lp withdraws')`